### PR TITLE
fix(flat-storage): Initiailze Flat Storage respecting the shards that the node will care about

### PR DESCRIPTION
**Pull request by nikurt**
Added a test reproducing this issue.

Steps to reproduce:
* Download an S3 snapshot
* Change config.json to track some shard in the next epoch but not in the current epoch.
* * Use `tracked_shard_schedule` or be a chunk-only producer.

Observe that the node tries to use flat storage to process chunks from shards that it will care about, but then it prints "FlatStorage not ready" and uses the usual Trie storage, resulting in `IncorrectStateRoot`.

### CODE REVIEW

**Comment by jakmeier** [comment]
@Longarithm this is about corner cases of flat storage. You and Anton are better suited to review this than me. Since he is OOO, I'll assign it to you.

File: pytest/tests/sanity/block_sync_flat_storage.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    height = boot_node.get_latest_block().height
+    if height >= EPOCH_LENGTH + 5:
+        break
+    if (len(keys) > 100 and random.random() < 0.2) or len(keys) > 1000:
+        key = keys[random.randint(0, len(keys) - 1)]
+        result = call_function(boot_node, 'read', key, nonce)
+    else:
+        key = random_u64()
+        keys.append(key)
+        result = call_function(boot_node, 'write', key, nonce)
```
Comment (Longarithm) [inline pytest/tests/sanity/block_sync_flat_storage.py:81]: Can be moved to function like `random_workload_until(node, block_height, keys, nonce) -> (keys, nonce)` to simplify code below.

File: pytest/tests/sanity/block_sync_flat_storage.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+node1.kill()
+apply_config_changes(node_dirs[1], {
+    "tracked_shards": [],
+    "tracked_shard_schedule": [[0], [0], [1, 2, 3]]
```
Comment (Longarithm) [inline pytest/tests/sanity/block_sync_flat_storage.py:86]: Unrelated, but I'd appreciate a comment somewhere in nearcore that `tracked_shard_schedule` overrides `tracked_shards` in the sense that only subset of shards is tracked (right?)

**Review by Longarithm — APPROVED** [review]

File: pytest/tests/sanity/block_sync_flat_storage.py (hunk 2 above)
Comment (nikurt) [reply pytest/tests/sanity/block_sync_flat_storage.py:86]: `tracked_shard_schedule` is only taken into account if `tracked_shards` is empty.
Added this comment to `ClientConfig`.

File: pytest/tests/sanity/block_sync_flat_storage.py (hunk 1 above)
Comment (nikurt) [reply pytest/tests/sanity/block_sync_flat_storage.py]: Done

**Pull request merged**